client/src/main/java/org/eclipse/hono/client/impl/CredentialsClientImpl.java

```diff
@@ -180,22 +203,29 @@ public static final String getTargetAddress(final String tenantId) {
                 .put(CredentialsConstants.FIELD_TYPE, type)
                 .put(CredentialsConstants.FIELD_AUTH_ID, authId)
                 .mergeIn(clientContext);
+        final TriTuple<CredentialsConstants.CredentialsAction, String, String> key = TriTuple.of(
```

Comment:
I think that we need to consider the client context in the key as well. IIRC the client context might actually be the *real* source of information used by the service to determine the data it puts into the response. For example with Kapua, all devices are using the same auth-id (conveyed in the username in an MQTT CONNECT packet) but all use different MQTT client IDs which are then used by the Credentials service implementation as the *real* auth-id and used to look up the password ...

Reply:
@sophokles73 `TriTuple.of(CredentialsConstants.CredentialsAction.get, String.format("%s-%s", type, authId), clientContext)` WDYT?

Reply:
I'd rather not keep a reference to the `JsonObject` in order to allow for it to be GCed more easily. Maybe we can use its hash code instead?

Reply:
I have used the hash code instead and pushed the changes.